ERCS/erc-7635.md

```diff
@@ -0,0 +1,521 @@
+depositing [ERC-20](./eip-20.md) or [ERC-721](./eip-721.md) tokens into the contract.
+
+Each MFT can act as a wallet to manage the slot assets associated with that MFT. These slot assets can be transferred to
+other MFTs or Ethereum addresses. When an MFT is transferred to a new Ethereum address, the slot tokens it manages are
+also transferred along with it, simplifying the asset migration process significantly.
+
+**Approving Mechanism**
+
+* This EIP provides additional slot authorization functionality in addition to being compatible with all authorization
+  rules of [ERC-721](./eip-721.md):
+    * `approve(uint256 _tokenId, uint256 _slotIndex, address _operator, uint256 valueOrNftId_)`:
+        * MUST revert if `_operator` is the same owner as the MFT
+        * An authorized operator of the current owner of an NFT
+    * For any approving function, the caller MUST be the owner or has been approved with a higher level of authority.
+
+**Transfer Mechanism**
+
+* The `transferFrom(uint256 _fromTokenId, uint256 _toTokenId, uint256 _slotIndex,uint256 _value)` function, SHOULD
+  indicate slot value transfers from one MFT to another MFT, in accordance with the rules below:
+
+    * MUST revert unless `msg.sender` is the owner of `_fromTokenId`, an authorized operator or an operator who has been
+      approved the whole token or at least `_value` of it.
+    * MUST revert if `_fromTokenId` or `_toTokenId` is zero token id or does not exist.
+    * MUST revert if `_slotIndex`  is zero id or does not exist.
+    * MUST revert if slots of `_fromTokenId` and `_toTokenId` do not match.
+    * MUST revert if `_value` exceeds the value of `_fromTokenId` or its allowance to the operator.
+    * MUST check for the `onERC7635Received` function if the owner of _toTokenId is a smart contract, if the function
+      exists, MUST call this function after the value transfer, MUST revert if the result is not equal to `0xde7a9e53`;
+    * MUST emit `TransferValue` event.
+
+* The `transferFrom(uint256 _fromTokenId, address _to, uint256 _slotIndex,uint256 _value)` function, which transfers
+  [ERC-20](./eip-20.md) or [ERC-721](./eip-721.md) value from one token ID to an address, SHOULD follow the rule below:
+
+    * MUST revert unless `msg.sender` is the owner of `_fromTokenId`, an authorized operator or an operator who has been
+      approved the whole token or at least `_value` of it.
+    * MUST revert if `_fromTokenId` is zero token id or does not exist.
+    * MUST revert if `_slotIndex`  is zero id or does not exist.
+    * MUST revert if `_to` is zero address.
+    * MUST revert if `_value` exceeds the value of `_fromTokenId` or its allowance to the operator.
+    * MUST check for the `onERC7635Received` function if the _to address is a smart contract, if the function exists, MUST
+      call this function after the value transfer, MUST revert if the result is not equal to `0xde7a9e53`;
+    * MUST emit `TransferValue` event.
```

Comment:
These bullets all introduce requirements, and so should be in the specification section.